src/main/java/it/pagopa/interop/probing/eservice/registry/reader/annotations/ValidateEnum.java

```diff
@@ -0,0 +1,67 @@
+/**************************************************************************
+*
+* Copyright 2023 (C) DXC
+*
+* Created on  : Feb 24, 2023
+* Author      : dxc technology
+* Project Name: interop-be-probing-eservice-registry-reader 
+* Package     : it.pagopa.interop.probing.eservice.registry.reader.annotations
+* File Name   : ValidateEnum.java
+*
+*-----------------------------------------------------------------------------
+* Revision History (Release )
+*-----------------------------------------------------------------------------
+* VERSION     DESCRIPTION OF CHANGE
+*-----------------------------------------------------------------------------
+** --/1.0  |  Initial Create.
+**---------|------------------------------------------------------------------
+***************************************************************************/
+package it.pagopa.interop.probing.eservice.registry.reader.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+import it.pagopa.interop.probing.eservice.registry.reader.annotations.validator.EnumValidator;
+
+
+/**
+ * The Interface ValidateEnum.
+ */
+@Target({ ElementType.FIELD, ElementType.PARAMETER })
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = EnumValidator.class)
+public @interface ValidateEnum {
+
+	 /**
+ 	 * Enum class.
+ 	 *
+ 	 * @return the class<? extends enum<?>>
+ 	 */
+ 	Class<? extends Enum<?>> enumClass();
+
+    /**
+     * Message.
+     *
+     * @return the string
+     */
+    String message() default "{uk.dds.ideskos.validator.ValidateString.message}";
```

Comment:
Remember to interpolate messages

Reply:
We already specify the class of the enum in the custom message in the DTO class. What specific useCase of the interpolation did you expect here?

Reply:
I thought you might want to change the default message and manage messages yourself